code/modules/paperwork/stamps.dm

```diff
@@ -12,6 +12,59 @@
 	item_color = "cargo"
 	pressure_resistance = 2
 	attack_verb = list("stamped")
+	var/stamp_color = "#264715"		//used for stamp marks, based on the item sprites.. not the on paper images(those are very bland)
+	var/stamp_description = "..."
+
+/obj/item/stamp/afterattack(atom/target, mob/user, proximity, params)
+	. = ..()
+	if(!ishuman(target) || !ishuman(user) || !proximity)
+		return
+
+	var/mob/living/carbon/human/H = target
+	var/attackedSide				//the stamp mark will appear on this side, NORTH = forwards, SOUTH = back, EAST = right, WEST = left
+	var/list/paramslist = params2list(params)
+	var/xOffset = text2num(paramslist["icon-x"]) - 16
+	var/yOffset = text2num(paramslist["icon-y"]) - 16
+
+	switch(H.dir)					//This makes the attackSide be the side that currently faces the 'camera'
+		if(NORTH)
+			attackedSide = SOUTH
+		if(SOUTH)
+			attackedSide = NORTH
+		if(EAST)
+			attackedSide = EAST
+		if(WEST)
+			attackedSide = WEST
+
+	//Here the base stamp sprite is loaded, in the dmi there are 3 different types of possible marks
+	//and for each type there is one state for each direction, although bizzare it going to have to be this way
+
+	var/icon/new_stamp_mark = icon('icons/effects/stamp_marks.dmi', "stamp[rand(1,3)]_[attackedSide]")
+	new_stamp_mark.Shift(EAST, xOffset)
+	new_stamp_mark.Shift(NORTH, yOffset)
+
+	//targetBaseIcon will be a picture of the human's base body underwear and uniform
+	//we will use it so stamp marks cannot be put on backpack or inhand items
+
+	var/mutable_appearance/targetBaseIcon = mutable_appearance()
+	targetBaseIcon.dir = H.dir
+	targetBaseIcon.overlays += H.overlays_standing[BODY_LAYER]
+	targetBaseIcon.overlays += H.overlays_standing[LIMBS_LAYER]
+	targetBaseIcon.overlays += H.overlays_standing[TAIL_LAYER]
+	targetBaseIcon.overlays += H.overlays_standing[UNDERWEAR_LAYER]
+	targetBaseIcon.overlays += H.overlays_standing[HAIR_LAYER]
+	targetBaseIcon.overlays += H.overlays_standing[UNIFORM_LAYER]
```

Comment:
Does one of these include clothes that go over jumpsuits? Like armor, jackets, hardsuits, etc.

Reply:
no, its only uniform.

Reply:
Currently as is they will show up in the same place regardless of what you are wearing and regardless if you change or remove clothes. This is to reduce complexity mostly, only uniform layer because I want to avoid floating stamp marks when you change backpacks or exosuits.

 I see an alternative as making a whole parallel code structure to the blood system but for stamp marks. When stamping we would check where the click lands on the human starting from the outer most layer working in, then adding a flag or data for each stamp mark onto the item of clothing or bodypart it does hit. Then render the mark when each layer of human is rendered, having a check for stamp marks when each respective layer is updated on their item/bodypart. Lastly have the stamping force a layer update for where it lands.

But that would be a very large undertaking that would have a much greater impact on the codebase as a whole. And to be honest I don't think I have a good enough grasp on it all to create an acceptable result.

Reply:
A project for you in the future maybe :stuck_out_tongue: